### PR TITLE
adds ability to sort professors by stats

### PR DESCRIPTION
**Pull request by atallahhezbor**
jQuery animations get a little choppy when there are many professors (try "all" on ECON 2020). Not sure if smoother way to do it other than sliding displayed panels up then sliding each down.

### CODE REVIEW

File: app/views/layouts/application.html.slim [hunk 1]
```diff
@@ -51,6 +51,16 @@ html
                   label.btn.btn-primary
                     input#all autocomplete="off" name="options" type="radio"
                     | All
```
Comment (aw3as) [inline app/views/layouts/application.html.slim]: more space in between button groups

File: app/views/courses/show_professors.html.slim [hunk 2]
```diff
@@ -5,48 +5,49 @@
...
-                    h4.rating-empty = "--"
-        .time
-          div style="float:right;"
-            .row.no-margin-row style="float:right;"
-              span.semester-subheader LAST TAUGHT
-            .row.no-margin-row
-              h4.semester = @professors_semester[professor.id]
\ No newline at end of file
+.container-fluid#browsing-content  
+  .prof-panel-container
```
Comment (aw3as) [inline app/views/courses/show_professors.html.slim]: why the additional nesting in prof-panel-container?

File: app/assets/javascripts/courses.js [hunk 3]
```diff
@@ -2,6 +2,91 @@ var ready;
...
+		var sortString = ".course-" + $(this).find('.active')[0].id;
+		// class of which panels are displayed (current semester or all)
+		var selectorString = $("#all").parent().hasClass("active") ? ".prof-panel.all" : ".prof-panel.current";
+		// how many panels there are (to know when to trigger the next animation)
+		var numPanels = $(selectorString).length;
+		var slidPanels = 0;		
+		// sorted list of professors
+		var profList = sortProfessors($(".prof-panel"), sortString);					
+		// slide up all the panels. then, on complete, add the sorted ones and slide down what is needed.
+		$(selectorString).slideUp(350, function() {
```
Comment (aw3as) [inline app/assets/javascripts/courses.js]: should disable slide animation when number is too high (performance)

File: app/assets/javascripts/courses.js [hunk 4]
```diff
@@ -2,6 +2,91 @@ var ready;
 
 ready = function() {
 
+	function sortProfessors(selector, attrName) {
+
+		return $($(selector).toArray().sort(function(a, b) {
```
Comment (aw3as) [inline app/assets/javascripts/courses.js]: better names! no a and b

File: app/assets/javascripts/courses.js [hunk 5]
```diff
@@ -2,6 +2,91 @@ var ready;
...
+
+		}));
+	}	
+	
+
+	// Sort professors by stats
+	$('#prof-sort').change(function() {		
+
+		// id of stat to sort by		
+		var sortString = ".course-" + $(this).find('.active')[0].id;
```
Comment (aw3as) [inline app/assets/javascripts/courses.js]: nest var statements like var x = 4, y = 3, z = 2

File: app/views/courses/show_professors.html.slim (hunk 2 above)
Comment (atallahhezbor) [inline app/views/courses/show_professors.html.slim]: felt bad doing $(".container-fluid").empty() in case more content is added

**Pull request merged**